test/contrib/postgres_test.py

```diff
@@ -124,6 +130,14 @@ def test_bulk_complete(self, mock_connect):
         ])
         self.assertFalse(task.complete())
 
+    def test_override_port(self):
+        output = DummyPostgresQueryWithPort(date=datetime.datetime(1991, 3, 24)).output()
+        self.assertEquals(output.port, 1234)
+
+    def test_port_encoded_in_host(self):
+        output = DummyPostgresQueryWithPortEncodedInHost(date=datetime.datetime(1991, 3, 24)).output()
+        self.assertEquals(output.port, '1234')
```

Comment:
This is rather unpleasant...

Reply:
Agreed.

The parsing is just done as a string split.. 

I'd suggest we leave this test here for this branch (for the sake of current state test success). Then address the fact that port could be a str if interpreted or an int/str if provided manually in a separate PR.

Reply:
Will merge then. Thanks for reviewing.

Reply:
Thank you for pushing back against the quick fix and taking over the more appropriate solutions! I greatly appreciate your involvement and look forward to the future aid in maintaining Luigi! :)